ipa-core/src/net/test.rs

```diff
@@ -91,30 +185,40 @@ pub fn server_config_https(
     }
 }
 
-#[derive(Default)]
 pub struct TestConfigBuilder {
-    ports: Option<[u16; 3]>,
+    /// Can be empty meaning that free ports should be obtained from the operating system.
```

Comment:
empty vector is a poor fit for this purpose. `Option<Vec<...>>` while more verbose, makes it much more clear that it is a different mode

Reply:
Makes sense